File: src/main/java/at/martinthedragon/ntm/main/CreativeTabs.java
package at.martinthedragon.ntm.main;

import at.martinthedragon.ntm.items.Items;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.ItemStack;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.api.distmarker.OnlyIn;

import javax.annotation.Nonnull;

public class CreativeTabs {
    public static final ItemGroup PARTS_TAB = new ItemGroup("ntm_parts") {
        @OnlyIn(Dist.CLIENT)
        @Nonnull
        public ItemStack createIcon() {
            return new ItemStack(Items.URANIUM_INGOT);
        }
    };
}

File: src/main/kotlin/at/martinthedragon/ntm/CreativeTabs.kt
package at.martinthedragon.ntm

import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack

enum class CreativeTabs(val itemGroup: ItemGroup) {
    Parts(
        object : ItemGroup("ntm_parts") {
            override fun makeIcon() = ItemStack(ModItems.uraniumIngot.get())
        }
    ),
    Items(
        object : ItemGroup("ntm_items") {
            override fun makeIcon() = ItemStack(ModItems.stoneFlatStamp.get()) // TODO
        }
    ),
    Templates(
        object : ItemGroup("ntm_templates") {
            override fun makeIcon() = ItemStack(ModItems.machineTemplateFolder.get())
        }
    ),
    Blocks(
        object : ItemGroup("ntm_blocks") {
            override fun makeIcon() = ItemStack(ModBlockItems.uraniumOre.get())
        }
    ),
    Machines(
        object : ItemGroup("ntm_machines") {
            override fun makeIcon() = ItemStack(ModBlockItems.safe.get()) // TODO
        }
    )
}
